src/components/modal/index.js

```diff
@@ -1,12 +1,17 @@
 import { h } from "preact";
-import { useState, useEffect } from "preact/hooks";
+import { useEffect } from "preact/hooks";
 
-import Button from "../button";
-import { Backdrop, ModalContainer } from "./styled";
+import {
+  Title,
+  Backdrop,
+  StyledImage,
+  TitleWrapper,
+  ModalContainer,
+} from "./styled";
 
-const Modal = ({ onRequestClose }) => {
+const Modal = ({ onRequestClose, imageSrc, title, isModalOpen, children }) => {
   const onKeyDown = (event) => {
-    if (event.keyCode === 27) {
+    if (event.keyCode === 27 && isModalOpen) {
```

Comment:
the event listener will be always there even if the modal is closed? 

In widget/index.js (https://github.com/bumi/area402-widget/pull/14/files#diff-128edf9b24e12b671c98d03163f11c4380faf0f88888c2c76a909cc786e167f2R32) it seems that we render the modal only if `isModalOpen` is set, but once it was rendered this event listener will be always active?

that means the event listener will still be there even if

Reply:
> the event listener will be always there even if the modal is closed?

No, it won't be. The event listener is cleaned up in the useEffect's return statement. In Functional components, the useEffect without any dependency array is equivalent to componentDidMount and when you add a return callback in the useEffect, it is equivalent to componentWillUnmount.

Reply:
ok, cool. 

what do you think about the `closeModal` option? instead of passing in the `isModalOpen` state and checking for that? ?

Reply:
I didn’t understand where you say that pass the `closeModal`  function as a prop instead of `isModalOpen`. Can you explain it?

Reply:
pr #25